src/common/components/layouts/pages/inspect/pdfDocument/PdfDocument.js

```diff
@@ -114,8 +118,17 @@ function PdfDocument(props) {
     }, [mapOfErrors, props.selectedCheck]);
     useEffect(() => {
         props.setSelectedCheck(activeBboxIndex);
+        if (activeBboxIndex && mapOfErrors[activeBboxIndex]) {
+            const { ruleIndex, checkIndex, location, context } = mapOfErrors[activeBboxIndex];
+            props.setSelectedCheckId(`${ruleIndex}:${checkIndex}:${location || context}`);
+            if (!props.expandedRule.includes(ruleIndex)) {
```

Comment:
Reuse the method passed in props instead of code duplication (See my very first comment)